etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/ruleengine/transformation/custom/CopyOrcOrderProviderToObrOrderProviderTest.groovy

```diff
@@ -0,0 +1,195 @@
+
+        transformClass = new CopyOrcOrderProviderToObrOrderProvider()
+    }
+
+    def "when both practitioner resources populated"() {
+        given:
+        final String EXPECTED_NPI = "1790743185"
+        final String EXPECTED_FIRST_NAME = "EUSTRATIA"
+        final String EXPECTED_LAST_NAME = "HUBBARD"
+        final String EXPECTED_NAME_TYPE_CODE = "NPI"
+        final String EXPECTED_IDENTIFIER_TYPE_CODE = null
+        final String FHIR_ORU_PATH = "../CA/007_CA_ORU_R01_CDPH_produced_UCSD2024-07-11-16-02-17-749_1_hl7_translation.fhir"
+
+        def bundle = createBundle(FHIR_ORU_PATH)
+        def serviceRequest = createServiceRequest(bundle)
+
+        expect:
+        evaluateOrc12Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+        evaluateObr16Values(serviceRequest, null, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, null, null)
+
+        when:
+        transformClass.transform(new HapiFhirResource(bundle), null)
+
+        then:
+        evaluateOrc12Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+        evaluateObr16Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+    }
+
+    def "when ORC-12 extension populated and OBR-16 extension not populated"() {
+        given:
+        final String EXPECTED_NPI = "1790743185"
+        final String EXPECTED_FIRST_NAME = "EUSTRATIA"
+        final String EXPECTED_LAST_NAME = "HUBBARD"
+        final String EXPECTED_NAME_TYPE_CODE = null
+        final String EXPECTED_IDENTIFIER_TYPE_CODE = "NPI"
+        final String FHIR_ORU_PATH = "../CA/017_CA_ORU_R01_CDPH_empty_obr16_UCSD2024-07-11-16-02-17-749_1_hl7_translation.fhir"
+
+        def bundle = createBundle(FHIR_ORU_PATH)
+        def serviceRequest = createServiceRequest(bundle)
+
+        expect:
+        // ORC12 values to copy
+        evaluateOrc12Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+
+        // OBR16 should not exist initially
+        def obr16Practitioner = getObr16ExtensionPractitioner(serviceRequest)
+        obr16Practitioner == null
+
+        when:
+        transformClass.transform(new HapiFhirResource(bundle), null)
+
+        then:
+        // ORC12 values should remain the same
+        evaluateOrc12Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+        // OBR16 values should be updated to match ORC12
+        evaluateObr16Values(serviceRequest, EXPECTED_NPI, EXPECTED_FIRST_NAME, EXPECTED_LAST_NAME, EXPECTED_NAME_TYPE_CODE, EXPECTED_IDENTIFIER_TYPE_CODE)
+    }
+
+    def "when ORC-12 extension not populated and OBR-16 extension is populated"() {
```

Comment:
`def "when ORC-12 extension not populated and OBR-16 extension is populated, there is no change"`